packages/xo-server/src/api/host.mjs

```diff
@@ -630,3 +635,46 @@ getIpmiSensors.params = {
 getIpmiSensors.resolve = {
   host: ['id', 'host', 'administrate'],
 }
+
+export async function getBiosInfo({ host }) {
+  const xapi = this.getXapi(host)
+
+  const biosData = await xapi.call('host.get_bios_strings', host._xapiRef)
+
+  const { 'bios-version': currentBiosVersion, 'system-product-name': hostServerName } = biosData
+
+  if (biosData['system-manufacturer']?.toLowerCase() !== '2crsi') {
+    return
+  }
+
+  if (!CACHE_2CRSI || !CACHE_2CRSI.has('servers')) {
+    const response = await fetch(
+      'https://pictures.2cr.si/Images_site_web_Odoo/Pages_produit/VATES-BIOS_BMC_last-version.json'
+    )
+    const json = await response.json()
+    const servers = keyBy(json[0]['2CRSi_Servers'], 'Server_Name')
+
+    if (CACHE_2CRSI) {
+      CACHE_2CRSI.set('servers', servers)
+    }
+  }
+
+  const parsedData = CACHE_2CRSI?.get('servers')
+
+  const serverData = parsedData?.[hostServerName]
+
+  if (serverData === undefined) {
+    return
+  }
+
+  const { 'BIOS-Version': latestBiosVersion, 'BIOS-link': biosLink } = serverData
+  const isUpToDate = currentBiosVersion === latestBiosVersion
+
+  return { currentBiosVersion, latestBiosVersion, biosLink, isUpToDate }
+}
```

Comment:
Thank you for removing this fro `@xen-orchestra/xapi`, now please move it again somewhere else (maybe `xo-server/xapi/index.js`, what do you think @MathieuRA) so this code can also be used outside the JSON-RPC API, like:

- internally by `xo-server`
- exposed one day in the REST API.